lib/Conversion/ExportVerilog/LegalizeNames.cpp

```diff
@@ -138,12 +138,21 @@ static void legalizeModuleLocalNames(HWModuleOp module,
   auto verilogNameAttr = StringAttr::get(ctxt, "hw.verilogName");
   // Legalize the port names.
   auto ports = module.getPortList();
-  for (auto [idx, port] : llvm::enumerate(ports)) {
-    auto newName = nameResolver.getLegalName(port.name);
-    if (newName != port.name.getValue()) {
-      module.setPortAttr(idx, verilogNameAttr, StringAttr::get(ctxt, newName));
-    }
-  }
+  auto setName = [&](size_t idx, Attribute name) {
```

Comment:
Why use a function here?  Just collect an array of the names.

Reply:
Wanted to save on constructing the array of names, also this iterates over the ports only once. Is it not worth saving the cost of an array construction here ?

Reply:
Fixed the API to just take an attribute array.